app/models/custom/poll/answer.rb

```diff
@@ -3,8 +3,23 @@
 class Poll::Answer
   belongs_to :answer, class_name: "Poll::Question::Answer", inverse_of: :answers
 
+  skip_validation :answer, :presence
   skip_validation :answer, :inclusion
 
-  validates :answer, inclusion: { in: ->(a) { a.question.question_answers }},
-                     unless: ->(a) { a.question.blank? }
+  validates :answer_id, presence: true,
+                        if: ->(a) { a.question&.single_choice? && a.question.mandatory_answer? }
+  validates :answer_id, inclusion: { in: ->(a) { a.question.question_answers.ids }},
+                        if: ->(a) { a.question&.single_choice? },
+                        allow_nil: true
+  validates :open_answer, presence: true,
+                          unless: ->(a) { a.question&.single_choice? },
+                          if: ->(a) { a.question&.mandatory_answer? }
+
+  def save_and_record_voter_participation
+    transaction do
+      touch if persisted?
+      save!
+      Poll::Voter.find_or_create_by!(user: author, poll: poll, origin: "web")
+    end
+  end
```

Comment:
This update you suggest requires more changes to work, and in my opinion, it does not worth the effort needed as soon as we know it works fine as it is now.

Reply:
Could you elaborate regarding which part isn't working with these changes? :thinking:

Reply:
If I were able to answer you I would already explained the reasons or did the changes instead, but I give up the research without understanding the real problem. Maybe it's not a big deal, but considering that we will probably remove the token thing from CONSUL, and if so, both kinds of customizations will be removed ... I decided to stop the fight. 

Maybe it's related to the fact that we assign `""` by default when the token is not present at [poll_helper] 🤔 (https://github.com/rockandror/innoviris-consul/blob/stable/app/helpers/polls_helper.rb#L16)

Reply:
Sorry,  I was asking about how to reproduce whatever isn't working, since everything I tested (which wasn't that much, to be honest :sweat_smile:) seemed to work fine.